src/test/kotlin/com/hibob/academy/dao/PetDaoTest.kt

```diff
@@ -17,32 +18,41 @@ class PetDaoTest @Autowired constructor (private val sql: DSLContext) {
     private val companyId = Random.nextLong()
     private val petDao = PetDao(sql)
 
+    // Test fields
+    private val ownerId1 = 101L
+    private val ownerId2 = 102L
+    private val petWithoutId1 = PetWithoutId (name = "Rex", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 1), ownerId1)
+    private val petWithoutId2 = PetWithoutId (name = "Whiskers", type = PetType.CAT, companyId = companyId, dateOfArrival = LocalDate.of(2024,2,2),ownerId2)
+    private val petWithoutId3 = PetWithoutId(name = "Rexi", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 2), ownerId1)
+    private val ownerlessPetWithoutId3 = PetWithoutId(name = "Rexi", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 2), null)
+
     @Test
     fun `insert pet test`() {
 
-        petDao.insertPet(name = "Rex", type = PetType.DOG, companyId = companyId, dateOfArrival = LocalDate.of(2024,1, 1))
+        petDao.insertPet(petWithoutId1)
```

Comment:
Instead of creating and the fetching by all fields to get the id - just set the insert into a val - which will contain also the created id and use this id later

Reply:
Amazing, would fix on next push.